### PR TITLE
[nightly] Build images for 2.1.0-preview1

### LINKED ISSUE #279: Begin producing 2.1 nightly images
Depends on microsoft/dotnet-nightly to begin consuming 2.1 daily builds of .NET Core.

cc @MichaelSimons

### PR DESCRIPTION
**Pull request by natemcmaster**
Fixes #279 

Changes:

`microsoft/aspnetcore-nightly`
 - No longer install the ASP.NET Core package runtime store. This has been replaced by the ASP.NET Core shared runtime
 - Base image is now `microsoft/dotnet-nightly:2.0-runtime-deps` instead of `2.0-runtime`. ASP.NET Core installers include the Microsoft.NETCore.App shared runtime

`microsoft/aspnetcore-build-nightly`
 - Remove bower and gulp 
 - Install NodeJS 8.9.3 - #334 
 -~~Install Git 2.16.1 (NanoServer only)~~ Removed git from the images.
 - Set `WORKDIR /root/code`. Helps with the common pitfall of trying to build the app in `/`, which fails b/c MSBuild tries to overwrite `/bin`.
 - Update base image to `microsoft/dotnet-nightly:2.1.300-preview1-sdk`
 - Don't run dotnet-restore when building the container. All packages should already exist in the NuGet fallback folder

Note: CI is expected to fail until microsoft/dotnet-nightly:2.1.300-preview1-sdk images are available.

cc @MichaelSimons @glennc

### CODE REVIEW

**Comment by MichaelSimons** [comment]
Required microsoft/dotnet-nightly changes are in https://github.com/dotnet/dotnet-docker-nightly/pull/546

File: 2.1/jessie/amd64/sdk/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM microsoft/dotnet-nightly:2.1.300-preview1-jessie
```
Comment (MichaelSimons) [inline 2.1/jessie/amd64/sdk/Dockerfile]: Base SDK tag used across platforms isn't consistent.  This particular usage isn't using the explicit SDK tag while others are.

File: 2.1/nanoserver-1709/amd64/sdk/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        Write-Host 'NODEJS CHECKSUM VERIFICATION FAILED!'; `
+        exit 1; `
+    }; `
+    `
+    Expand-Archive node.zip -DestinationPath nodejs-tmp; `
+    Move-Item nodejs-tmp/node-v${env:NODE_VERSION}-win-x64 nodejs; `
+    Remove-Item -Force node.zip; `
+    Remove-Item -Force nodejs-tmp
+
+## https://github.com/git-for-windows/git/releases
```
Comment (MichaelSimons) [inline 2.1/nanoserver-1709/amd64/sdk/Dockerfile]: I would like to know more on the motivation for adding git.  Have you been getting requests for this?  I haven't heard of any which is why I ask.  Should this be pushed to the base image?

File: 2.1/nanoserver-sac2016/amd64/sdk/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# escape=`
+FROM microsoft/dotnet-nightly:2.1.300-preview1-sdk-nanoserver-sac2016
+
+# set up environment
+ENV ASPNETCORE_URLS http://+:80
+ENV NODE_VERSION 8.9.4
```
Comment (MichaelSimons) [inline 2.1/nanoserver-sac2016/amd64/sdk/Dockerfile]: Looks to be defined multiple times.

File: 2.1/nanoserver-1709/amd64/sdk/Dockerfile (hunk 2 above)
Comment (natemcmaster) [reply 2.1/nanoserver-1709/amd64/sdk/Dockerfile]: Now that I think about it, we can probably remove this. I believe installing it was a requirement to make bower work. We're not installing bower anymore.

File: 2.1/jessie/amd64/sdk/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,25 @@
+FROM microsoft/dotnet-nightly:2.1.300-preview1-sdk-jessie
+
+# set up environment
+ENV ASPNETCORE_URLS http://+:80
+
+# Ensure packages required for standalone apps are pre-fetched
+RUN mkdir /tmp/warmup && cd /tmp/warmup \
```
Comment (MichaelSimons) [inline 2.1/jessie/amd64/sdk/Dockerfile]: I find the mixture of multi-command lines  (e.g. && cd / && rm -rf /tmp/warmup) and single lines to be a little confusing.

File: 2.1/nanoserver-1709/amd64/sdk/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    Remove-Item -Force nodejs-tmp
+
+# Build image
+FROM microsoft/dotnet-nightly:2.1.300-preview1-sdk-nanoserver-1709
+
+# Set up environment
+ENV ASPNETCORE_URLS http://+:80
+
+COPY --from=installer-env ["nodejs", "C:\\Program Files\\nodejs"]
+USER ContainerAdministrator
```
Comment (MichaelSimons) [inline 2.1/nanoserver-1709/amd64/sdk/Dockerfile]: Nit - inconsistent use of comments across Dockerfiles explaining the use of ContainerAdministrator

File: 2.1/nanoserver-1709/amd64/sdk/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        exit 1; `
+    }; `
+    `
+    Expand-Archive node.zip -DestinationPath nodejs-tmp; `
+    Move-Item nodejs-tmp/node-v${env:NODE_VERSION}-win-x64 nodejs; `
+    Remove-Item -Force node.zip; `
+    Remove-Item -Force nodejs-tmp
+
+# Build image
+FROM microsoft/dotnet-nightly:2.1.300-preview1-sdk-nanoserver-1709
```
Comment (MichaelSimons) [inline 2.1/nanoserver-1709/amd64/sdk/Dockerfile]: Curious why the difference in the order the components are installed between Linux and windows.

File: manifest.json [hunk 7]
```diff
@@ -118,21 +118,21 @@
           },
           "platforms": [
             {
-              "dockerfile": "2.1/stretch/runtime",
+              "dockerfile": "2.1/stretch/amd64/runtime",
```
Comment (MichaelSimons) [inline manifest.json]: The readme should also get updated to correctly reference the new Dockerfile locations.

File: 2.1/nanoserver-1709/amd64/sdk/Dockerfile (hunk 6 above)
Comment (natemcmaster) [reply 2.1/nanoserver-1709/amd64/sdk/Dockerfile]: No specific reason. I'll update to be consistent.

File: manifest.json (hunk 7 above)
Comment (natemcmaster) [reply manifest.json]: Updated.

**Comment by natemcmaster** [comment]
@MichaelSimons updated with cleanup according to your feedback.

File: test/Dockerfile.testrunner.linux [hunk 8]
```diff
@@ -1,15 +1,12 @@
 # Use this Dockerfile to create a Linux testrunner image
 #     docker build -t testrunner -f ./test/Dockerfile.testrunner.linux .
-#     docker run --rm -v /var/run/docker.sock:/var/run/docker.sock testrunner powershell -File test/test.ps1 <test.ps1 args>
+#     docker run --rm -v /var/run/docker.sock:/var/run/docker.sock testrunner test/test.ps1 <test.ps1 args>
 
-FROM ubuntu:16.04
+FROM microsoft/powershell:ubuntu16.04
```
Comment (MichaelSimons) [inline test/Dockerfile.testrunner.linux]: You may consider using a prebuilt image here - debian-stretch-docker-testrunner-63f2145-20184325094343.  The source for it is at https://github.com/dotnet/dotnet-buildtools-prereqs-docker/blob/master/src/debian/stretch/docker-testrunner/Dockerfile.  Would save some time not having to build this entire thing every time.  This is being used in dotnet-docker-nightly - https://github.com/dotnet/dotnet-docker-nightly/pull/538

Comment (natemcmaster) [reply test/Dockerfile.testrunner.linux]: I'll give that a shot

Comment (natemcmaster) [reply test/Dockerfile.testrunner.linux]: Works great! Thanks for pointing that out @MichaelSimons

**Pull request merged**